woocommerce/google-listings-and-ads
# Externalize `Dashicon`, `Icon`, `Notice` and `Tip` components imported from `@wordpress/components`

### Changes proposed in this Pull Request:

It's part of #1833

- Externalize `Dashicon`, `Icon`, `Notice`, and `Tip` components.
- Opt-in the `@wordpress/primitives` package to use the DEWPed one.
   - This is needed due to `Icon` compatibility. The `icon.type === SVG` comparison result needs to be `true`.
   - [Reference](https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4019.2.2/packages/components/src/icon/index.tsx#L84)
- Extra changes:
   - Fix missing CSS selectors when renaming `AppTooltip` CSS name in [#1901](https://github.com/woocommerce/google-listings-and-ads/pull/1901).
   - Remove an unused component `DismissibleNotice` that was missed to be removed in [#1889](https://github.com/woocommerce/google-listings-and-ads/pull/1889).

### Screenshots:

#### 📷 `Dashicon` component on the Product Feed page

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/c510fff2-df16-4f58-b151-936ac960d6e4)

#### 📷 `Icon` and `Notice` components on the Get Started page

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/e3e01fc3-5b88-40b2-bc85-ad6fc85bbfbd)

#### 📷 `Tip` component on the Get Started page

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/b4c473fc-4c60-4945-b083-3a190015c5bd)

#### 📷 The edit icon color was corrected to gray

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/d18049e7-483b-406e-ae5b-aaf2db4a6c73)

#### 📷 The CSS style of "N products" text wrapped by `AppTooltip` was corrected

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/706c7641-8f53-48fe-8173-7e3550ccac07)

### Detailed test instructions:

#### 💡 It would be easier to test this PR by making a few changes locally:
- Make [is_language_supported](https://github.com/woocommerce/google-listings-and-ads/blob/f61075c24c3df8ca400658909e517d5141b1c0a9/src/MerchantCenter/MerchantCenterService.php#L147-L160) return false.
- Add `return <IssuesSolved />;` to [this line](https://github.com/woocommerce/google-listings-and-ads/blob/f61075c24c3df8ca400658909e517d5141b1c0a9/js/src/product-feed/issues-table-card/issues-table-data.js#L47).

#### Testing:

1. `npm start`
2. Go to the Get Started page.
3. Check if the `Icon`, `Notice` and `Tip` components are shown correctly.
4. Go to step 4 of the onboarding flow.
5. Check if the style of the "N products" text is corrected as the screenshot above.
6. Go to the Product Feed page.
7. Check if the `Dashicon` components are shown correctly.
8. Check if the color of edit icon on the Product Feed table is corrected to gray.
9. Search the codebase to check if all these components are imported from `'extracted/@wordpress/components'`.

### Changelog entry
